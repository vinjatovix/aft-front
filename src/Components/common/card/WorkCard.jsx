import React from "react";
import PropTypes from "prop-types";
import { ListItem } from "../ListItem";

export const WorkCard = ({ data }) => {
  return (
    <ul className="work-list">
      <ListItem
        name="Intérprete"
        children={<p>by: {data.metadata.createdBy}</p>}
        liKey={data.metadata.createdBy}
      />
      <ListItem
        name="Obra"
        children={<p>Obra: {data.character.book.name}</p>}
        liKey={data.character.book._id}
      />
      <ListItem
        name="Personaje"
        children={<p>Personaje: {data.character.name}</p>}
        liKey={data.character._id}
      />
      <ListItem
        name="Escena"
        children={<p>Escena: {data.scene.name}</p>}
        liKey={data.scene._id}
      />
      <ListItem
        name="Descripción"
        children={<p>Descripción: {data.description}</p>}
      />
    </ul>
  );
};

WorkCard.propTypes = {
  data: PropTypes.object.isRequired,
};
