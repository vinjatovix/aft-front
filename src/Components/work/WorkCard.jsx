import React from "react";
import PropTypes from "prop-types";

import { ListItem } from "../common/ListItem";

export const WorkCard = ({ data }) => {
  return (
    <ul className="work-list">
      <ListItem
        name="Intérprete"
        children={<p>by: {data.metadata.createdBy}</p>}
        liKey={data.metadata.createdBy}
      />
      <ListItem
        name="Obra"
        children={<p>Obra: {data.character.book.name}</p>}
        liKey={data.character.book._id}
      />
      <ListItem
        name="Personaje"
        children={<p>Personaje: {data.character.name}</p>}
        liKey={data.character._id}
      />
      <ListItem
        name="Escena"
        children={<p>Escena: {data.scene.name}</p>}
        liKey={data.scene._id}
      />
      <ListItem
        name="Descripción"
        children={<p>Descripción: {data.description}</p>}
      />
    </ul>
  );
};

WorkCard.propTypes = {
  data: PropTypes.shape({
    character: PropTypes.shape({
      book: PropTypes.shape({
        name: PropTypes.string.isRequired,
      }).isRequired,
      name: PropTypes.string.isRequired,
    }).isRequired,
    description: PropTypes.string,
    metadata: PropTypes.shape({
      createdBy: PropTypes.string.isRequired,
    }).isRequired,
    scene: PropTypes.shape({
      name: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};
