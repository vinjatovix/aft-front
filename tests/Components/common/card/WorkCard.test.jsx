import { render } from "@testing-library/react";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { WorkCard } from "../../../../src/Components/work/WorkCard";
import mockActions from "../../../fixtures/mockActions";

describe("WorkCard Component", () => {
  const data = {
    _id: "5e9f8f8f8f8f8f8f8f8f8f8",
    description: "DescripciÃ³n",
    metadata: {
      createdBy: "user",
    },
    character: {
      book: {
        name: "Obra",
      },
      name: "Personaje",
    },
    scene: {
      name: "Escena",
    },
  };

  it("should match snapshot", async () => {
    const { container } = render(
      <Router>
        <WorkCard
          data={data}
          isEditor={true}
          actions={mockActions}
          token="token"
        />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
});
