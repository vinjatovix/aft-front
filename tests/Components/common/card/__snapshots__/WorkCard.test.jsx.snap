// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkCard Component should match snapshot 1`] = `
<div>
  <ul
    class="work-list"
  >
    <li
      class=""
      data-testid="Intérprete"
    >
      <p>
        by: 
        user
      </p>
    </li>
    <li
      class=""
      data-testid="Obra"
    >
      <p>
        Obra: 
        Obra
      </p>
    </li>
    <li
      class=""
      data-testid="Personaje"
    >
      <p>
        Personaje: 
        Personaje
      </p>
    </li>
    <li
      class=""
      data-testid="Escena"
    >
      <p>
        Escena: 
        Escena
      </p>
    </li>
    <li
      class=""
      data-testid="Descripción"
    >
      <p>
        Descripción: 
        Descripción
      </p>
    </li>
  </ul>
</div>
`;
